ui/src/telegrafs/components/TelegrafConfigOverlay.tsx

```diff
@@ -1,102 +1,109 @@
 // Libraries
 import React, {PureComponent} from 'react'
 import {connect} from 'react-redux'
+import {withRouter, WithRouterProps} from 'react-router'
 
 // Components
 import {ErrorHandling} from 'src/shared/decorators/errors'
 import TelegrafConfig from 'src/telegrafs/components/TelegrafConfig'
-import {ComponentColor, Button} from '@influxdata/clockface'
+import {
+  ComponentColor,
+  Button,
+  RemoteDataState,
+  SpinnerContainer,
+  TechnoSpinner,
+} from '@influxdata/clockface'
 import {Overlay} from 'src/clockface'
 
 // Utils
 import {downloadTextFile} from 'src/shared/utils/download'
 
-// Constants
-import {getTelegrafConfigFailed} from 'src/shared/copy/v2/notifications'
-
-// APIs
-import {client} from 'src/utils/api'
-
-// Actions
-import {notify as notifyAction} from 'src/shared/actions/notifications'
-
 // Types
 import {AppState} from 'src/types'
-
-interface OwnProps {
-  visible: boolean
-  onDismiss: () => void
-}
+import {ITelegraf as Telegraf} from '@influxdata/influx'
 
 interface StateProps {
-  telegrafConfigName: string
-  telegrafConfigID
+  telegraf: Telegraf
+  status: RemoteDataState
+  telegrafConfig: string
 }
 
-interface DispatchProps {
-  notify: typeof notifyAction
-}
-
-type Props = OwnProps & StateProps & DispatchProps
+type Props = StateProps & WithRouterProps
 
 @ErrorHandling
-export class TelegrafConfigOverlay extends PureComponent<Props> {
+class TelegrafConfigOverlay extends PureComponent<Props> {
   public render() {
-    const {visible, onDismiss, telegrafConfigName} = this.props
-
-    return (
-      <Overlay visible={visible}>
-        <Overlay.Container maxWidth={1200}>
-          <Overlay.Heading
-            title={`Telegraf Configuration - ${telegrafConfigName}`}
-            onDismiss={onDismiss}
-          />
-
-          <Overlay.Body>
-            <div className="config-overlay">
-              <TelegrafConfig />
-            </div>
-          </Overlay.Body>
-          <Overlay.Footer>
-            <Button
-              color={ComponentColor.Secondary}
-              text="Download Config"
-              onClick={this.handleDownloadConfig}
-            />
-          </Overlay.Footer>
-        </Overlay.Container>
-      </Overlay>
-    )
+    return <>{this.overlay}</>
   }
 
-  private handleDownloadConfig = async () => {
-    try {
-      const config = await client.telegrafConfigs.getTOML(
-        this.props.telegrafConfigID
-      )
-      downloadTextFile(
-        config,
-        `${this.props.telegrafConfigName || 'config'}.toml`
+  private get overlay(): JSX.Element {
+    const {telegraf, status} = this.props
+
+    if (status === RemoteDataState.Done) {
```

Comment:
because of this check, we will never see the spinner below. How do you feel about removing this check so the spinner can do its thing?

Reply:
The check for status is needed here because if the telegraf list is not done loading, it throws an error on telegraf name that is being displayed in the overlay heading.